### ISSUE TITLE
Using Bluetooth to retrieve data from EZ1?

### ISSUE THREAD

**Issue by bluepuma77** [SonnenladenGmbH/APsystems-EZ1-API#39]
Got an EZ1 from Sonnenladen, it seems the smartphone app can fetch data with "direct connect" via Bluetooth. Is it possible to use the BT method instead of using WiFi/IP? Does the Bluetooth API need to "connect", does it need authentication, is it BT LE, is it just broadcasting data?

Would be interested in using an ESP32 with MicroPython/CircuitPython to fetch data from EZ1 and forward it via LoRaWan.

**Comment by mawoka-myblock** [comment]
> Got an EZ1 from Sonnenladen, it seems the smartphone app can fetch data with "direct connect" via Bluetooth. Is it possible to use the BT method instead of using WiFi/IP? Does the Bluetooth API need to "connect", does it need authentication, is it BT LE, is it just broadcasting data?
> 
> Would be interested in using an ESP32 with MicroPython/CircuitPython to fetch data from EZ1 and forward it via LoRaWan.

Sounds like a good plan, but the problem is that the inverter doesn't transmit any bluetooth data after a successful wifi connection and as I don't want to change a running system, I don't have any option to test things out. I'll keep this issue open as maybe some people could help me in this regard.

As you seem technical and if you would like to assist me, you could get the "GATTBrowser" app by Renesas on your android phone and connect to your EZ1 prior to wifi setup, but after the normal setup in the app. Then, I'd love to receive a screenshot of all the attributes and, if possible, try "reading" every readable attribute. That'd be pretty much all the information I'd need for developing the bluetooth API.

**Comment by ccvca** [comment]
I've had exactly same idea, as the wifi connection is not really working. (Need a Wifi Repeater, which likes to crash form time to time)

At least in my case, bluetooth is still active, event if the Wifi is connected and the local API is used (Using the APsystem App and direct connect). But I've never setup the cloud, so maybe bluetooth is gone after logging in with a cloud account. I'll try to get some data. A BLE-Scanner can at least read the name.

**Comment by mawoka-myblock** [comment]
Yea, I've also tried the BLE scanner and it also could only read the name, not more unfortunately.

**Comment by ccvca** [comment]
Yeah, but there are some more services and a possibility to write. Maybe some kind of "code" needs to be written first.... At least I can't find a clue, that there is any Bluetooth classic involved (at least Android does not show it to me).

**Comment by mawoka-myblock** [comment]
Pretty much the same for me. But as you can read in my original reply, I'm not that interested in that and also don't have the skills for Bluetooth reverse engineering (yet).

**Comment by bensoibj** [comment]
I tried to scan and connect to the EZ1 with the Bluetooth scanner “[toolBLEx](https://github.com/emericg/toolBLEx)”. My device runs with firmware 1.7.5 and was never connected via Wifi, until now only via Bluetooth.

It seems that only a single connection to the EZ1 is possible at the same time. When the app is connected, the scanner does not see the device. When the scanner is connected, the app can’t connect. Once connected to the EZ1, it stops to advertise via Bluetooth.

Attached two screenshots while scanning and when connected. Here is the scanner data in text format:
```
Device Name: EZ1_E07010027956
Device MAC: 80:64:6F:63:2F:7A
Device MAC manufacturer: Espressif Inc.

First seen: Sun Oct 5 12:16:32 2025
Last seen: Sun Oct 5 12:45:42 2025

Service(s) advertised:
- {00001800-0000-1000-8000-00805F9B34FB}
- {00001801-0000-1000-8000-00805F9B34FB}
- {0000FFFE-0000-1000-8000-00805F9B34FB}

Advertising interval: 116700 ms
> No advertisement packets.

Service Name: "Generic Access"
Service UUID: {00001800-0000-1000-8000-00805F9B34FB}
Characteristic Name: GAP Device Name - UUID: {00002A00-0000-1000-8000-00805F9B34FB} - Properties: Read - Value: 0x455A315F453037303130303237393536
Characteristic Name: GAP Appearance - UUID: {00002A01-0000-1000-8000-00805F9B34FB} - Properties: Read - Value: 0x0000

Service Name: "Generic Attribute"
Service UUID: {00001801-0000-1000-8000-00805F9B34FB}
Characteristic Name: GATT Service Changed - UUID: {00002A05-0000-1000-8000-00805F9B34FB} - Properties: Indicate - Value: <none>

Service Name: "Unknown Service"
Service UUID: {0000FFFE-0000-1000-8000-00805F9B34FB}
Characteristic Name: Unknown Characteristic - UUID: {0000FF0A-0000-1000-8000-00805F9B34FB} - Properties: Write - Value: <none>
Characteristic Name: Unknown Characteristic - UUID: {0000FF0B-0000-1000-8000-00805F9B34FB} - Properties: Read / Notify - Value: 0x00
```

I guess the communication is done via the two characteristics `0000FF0A-0000-1000-8000-00805F9B34FB` and `0000FF0B-0000-1000-8000-00805F9B34FB`. Maybe the first, writeable one is for sending a request, and the second one returns the result afterwards. 
I tried to send something to the EZ1 via the first characteristic, but none of my trials (“getOutputData”, `0x00`, `0x01`) succeeded, the device was rather immediately disconnected. I will continue to test when I have more time…

### Bluetooth Scan filtered
<img width="2884" height="1545" alt="Bluetooth Scan filtered" src="https://github.com/user-attachments/assets/77ab8ad0-5321-4738-a11a-86c8a4f67ad3" />

### Bluetooth connected
<img width="906" height="1672" alt="Bluetooth connected" src="https://github.com/user-attachments/assets/25490e63-ccd2-4479-9a83-f5b14c1db939" />
